javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala

```diff
@@ -0,0 +1,43 @@
+package java.util.concurrent.atomic
+
+import scala.collection.mutable.ArrayBuffer
+
+class AtomicReferenceArray[E <: AnyRef](
+    length: Int) extends Serializable {
+
+  def this(array: Array[E]) = {
+    this(array.size)
+    inner = ArrayBuffer[E](array: _*)
```

Comment:
Instead of reassigning `inner` to another thing (and requiring it to be a `var`), use `System.arraycopy` to copy the elements to the already allocated `inner`.